src/govuk/settings/_typography-font.scss

```
@@ -24,6 +26,17 @@ $govuk-use-legacy-font: if(
   false
 ) !default;
 
+// Only show the deprecation warning if user is setting $govuk-use-legacy-font
+// manually instead of automatically via compatibility variables
+@if $govuk-use-legacy-font == true and
+  $govuk-compatibility-govukfrontendtoolkit == false and
+  $govuk-compatibility-govuktemplate == false and
+  $govuk-compatibility-govukelements == false {
+  @warn "$govuk-use-legacy-font is deprecated. " +
+    "From version 5.0, we will only support the latest version of the " +
+    "GDS Transport font, included with govuk-frontend";
```

For brevity and clarity:

```suggestion
    "From version 5.0, GOV.UK Frontend will only support the included version " + 
    "of GDS Transport"
```